### ISSUE TITLE
explainer: unpack packed storage slots

### ISSUE THREAD

**Issue by simon-jentzsch** [corpus-core/colibri-stateless#380]
## Context

Several variables can share one 32-byte storage word. Example UniswapV2Pair: `reserve0` (uint112), `reserve1` (uint112), `blockTimestampLast` (uint32) live in one slot. The prompt currently shows the whole word under the first variable:

```
0x9b8d...e2f7: reserve0 (uint112): 48191879296563733225818505312093938157060312406052357127986980923455007860267 -> ...
```

The value is far too large for a uint112. This is **confidently wrong** data — worse than an unresolved slot, because the model learns to trust wrong labels. V2 pairs are among the most common contracts, so a large share of the training data is affected.

## Proposal

Use `offset` and `numberOfBytes` from the storage layout (`SolidityStorageEntry.offset`, `SolidityStorageType.numberOfBytes`) to split the word into its members, and print only the members that actually changed:

```
[s4] UniswapV2Pair: reserve0 (uint112): 4481305370003744299 -> 4515840257591060011
[s4] UniswapV2Pair: reserve1 (uint112): 82828854080223465119844462 -> 82197306331417110159722606
```

(Both member lines keep the ID of the underlying slot change.)

## Tasks

- [ ] `storage.ts`: resolve all layout entries sharing the slot, return one entry per member with extracted before/after values
- [ ] Also for packed struct members and packed mapping values
- [ ] Without layout information: never label the whole word with one variable name — print `slot N [unresolved]` instead
- [ ] Triage over the training set: count affected examples before/after

## Acceptance criteria

- UniswapV2Pair `Sync` swap shows `reserve0` / `reserve1` with plausible uint112 values
- No resolved variable is printed with a value exceeding its type width (assertion in builder)
